src/charm.py

```diff
@@ -213,6 +253,10 @@ def _missing_relations(self) -> List[str]:
                 missing_relations.append(relation)
         return missing_relations
 
+    @property
+    def _webui_data_is_available(self) -> bool:
```

Comment:
Not sure about the name here. Perhaps it would be more clear to name it `_webui_url_is_available` as we are not checking for data, but the presence of the url

Reply:
We were considering the URL as a data by following the same context like NRF. Here NRF data is also checking for NRF URL. But, I am fine with changing data with URL.

Reply:
Property is renamed as ` _webui_url_is_available`.